cassandra/cqltypes.py

```diff
@@ -450,6 +450,8 @@ def serialize(dec, protocol_version):
             try:
                 sign, digits, exponent = Decimal(dec).as_tuple()
             except Exception:
+                if Decimal(dec).is_nan():
```

Comment:
We're instantiating another instance the same value we have in the try block above here.  Would be nice if there was some way to use a single value in both places, although we'll want to make sure any errors that might be raised by that constructor are handled.